### PR TITLE
Cef keyinput modifiers

### PR DESCRIPTION
**Pull request by shaneharris**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -81,7 +81,7 @@
     "window-fetch": "0.0.9",
     "window-selector": "0.0.5",
     "window-text-encoding": "0.0.2",
-    "window-worker": "0.0.97",
+    "window-worker": "0.0.98",
```
Comment (modulesio) [inline package.json]: Probably shouldn't include this in this PR. There is another PR where it's held up on multi-platform testing.

File: deps/exokit-bindings/browser/src/browser.cpp [hunk 2]
```diff
@@ -536,19 +556,44 @@ NAN_METHOD(Browser::SendMouseWheel) {
   }
 }
 
+int GetKeyModifiers(Local<Object> modifiersObj){
+  int modifiers = 0;
+  if(modifiersObj->Get(JS_STR("shiftKey"))->BooleanValue()){
+    modifiers |= EVENTFLAG_SHIFT_DOWN;
+	std::cout << "shiftKey" << std::endl;
```
Comment (modulesio) [inline deps/exokit-bindings/browser/src/browser.cpp]: Will have to remove these.

File: examples/browser.html [hunk 3]
```diff
@@ -171,7 +171,7 @@
     const coord = planeMesh.projectMouse(e.clientX, e.clientY);
 
     if (coord) {
-      iframe.sendMouseDown(coord.x, coord.y);
+      iframe.sendMouseDown(coord.x, coord.y,0); //button 0 - left , 1 middle, 2 - right
```
Comment (modulesio) [inline examples/browser.html]: Should get the actual button from the event.

File: deps/exokit-bindings/browser/src/browser.cpp [hunk 4]
```diff
@@ -578,18 +627,25 @@ NAN_METHOD(Browser::SendKeyUp) {
...
 NAN_METHOD(Browser::SendKeyPress) {
-  if (info[0]->IsNumber()) {
+  // Nan::HandleScope scope;
+  if (info[0]->IsNumber() && info[1]->IsObject()) {
     Browser *browser = ObjectWrap::Unwrap<Browser>(info.This());
     int key = info[0]->Uint32Value();
+	
+	Local<Object> modifiersObj = Local<Object>::Cast(info[1]);
+	int modifiers = GetKeyModifiers(modifiersObj);
+	if(modifiersObj->Get(JS_STR("shiftKey"))->BooleanValue()){
```
Comment (modulesio) [inline deps/exokit-bindings/browser/src/browser.cpp]: This is the current bug: CEF not handling shift.

File: deps/exokit-bindings/browser/src/browser.cpp (hunk 2 above)
Comment (shaneharris) [reply deps/exokit-bindings/browser/src/browser.cpp]: will remove now

File: package.json (hunk 1 above)
Comment (shaneharris) [reply package.json]: yep sorry about that one

File: deps/exokit-bindings/browser/src/browser.cpp (hunk 4 above)
Comment (shaneharris) [reply deps/exokit-bindings/browser/src/browser.cpp]: this does hit this block ok, so this part seems right

File: examples/browser.html [hunk 5]
```diff
@@ -197,21 +197,21 @@
...
+      iframe.sendKeyUp(e.keyCode,{shiftKey:e.shiftKey,ctrlKey:e.ctrlKey,altKey:e.altKey});
     }
   });
   window.addEventListener('keypress', e => {
     if (focused) {
       if (e.keyCode === 114 && e.ctrlKey) { // Ctrl-R
         iframe.reload();
       } else {
-        iframe.sendKeyPress(e.keyCode);
+        iframe.sendKeyPress(e.keyCode,{shiftKey:false,ctrlKey:false,altKey:false});
```
Comment (modulesio) [inline examples/browser.html]: Wait why false?

**Pull request closed without merging**